submissions/antonkottans/friends-app/script.js

```diff
@@ -0,0 +1,582 @@
+const pipe = (...arrayOfFunctions) => {
+  if (arrayOfFunctions.length === 0) return null;
+  else if (arrayOfFunctions.length === 1) return arrayOfFunctions[0]();
+  return arrayOfFunctions.reduceRight(
+    (recur, func) =>
+      (...args) =>
+        recur(func(...args))
+  )();
+};
+
+const nodes = {
+  paginationRadioButtons: null,
+  containerForFriendsNode: null,
+  radioNavContainer: null,
+};
+
+const constants = {
+  MIN_AGE: 0,
+  MAX_AGE: 200,
+};
+
+const state = {
+  loading: false,
+  allFriends: [],
+  friendsFilter: {
+    sortByAge: null,
+    sortByName: null,
+    ageFilterRange: { min: constants.MIN_AGE, max: constants.MAX_AGE },
+    nameFilter: "",
+    sexFilter: "all",
+  },
+};
+
+const resetFilters = () => {
+  state.friendsFilter.sortByAge = null;
+  state.friendsFilter.sortByName = null;
+  state.friendsFilter.ageFilterRange = {
+    min: constants.MIN_AGE,
+    max: constants.MAX_AGE,
+  };
+  state.friendsFilter.nameFilter = "";
+  state.friendsFilter.sexFilter = "all";
+};
+
+const cardSize = {
+  width: 200,
+  height: 268,
+};
+
+const calcPagesAmount = (
+  containerHeight,
+  containerWidth,
+  amountOfElements,
+  gridGap = 10
+) => {
+  let columns = Math.trunc(containerWidth / cardSize.width) || 1;
+  if (containerWidth % cardSize.width <= gridGap * (columns - 1)) columns--;
+  let rows = Math.trunc(containerHeight / cardSize.height) || 1;
+  if (containerHeight % cardSize.height <= gridGap * (rows - 1)) rows--;
+  let amountPerPage = columns * rows;
+  let amountOfPages = Math.ceil(amountOfElements / amountPerPage);
+  return { amountPerPage, amountOfPages };
+};
+
+const getNearestNaturalNumbers = (
```

Comment:
Do you have float numbers in your work?)

Reply:
no, but this function doesn`t return zero or negative numbers.